Fixes two issue:
- Fixes build issue related to libreadline
- Allows build with ICU 77 (adds now needed -licuuc)

--- mozjs-78.4.0/js/moz.configure
+++ mozjs-78.4.0/js/moz.configure
@@ -919,12 +919,12 @@
 
 has_readline = check_symbol(
     "readline",
-    flags=["-lreadline"],
+    flags=["-lreadline", "-lcurses"],
     when="--enable-readline",
     onerror=lambda: die("No system readline library found"),
 )
 
-set_config("EDITLINE_LIBS", ["-lreadline"], when=has_readline)
+set_config("EDITLINE_LIBS", ["-lreadline", "-lcurses", "-licuuc"], when=has_readline)
 
 
 @depends("--enable-readline", editline, when=editline)
